File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        # Start from the last digit and move backwards
        for i in range(len(digits) -1, -1, -1):
            # If digit is less than 9, just add 1 and return
            if digits[i] < 9:
                digits[i] += 1
                return digits

            # if digit is 9, it becomes 0 and carry the 1 to the next left digit
            digits[i] = 0

        # if all digits were 9, we need a leading 1
        return [1] + digits        

File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        // walk backwards, if not 9 increment then return 
        for (int index = digits.length -1; index >= 0; index--) {
            if (digits[index] != 9) {
                digits[index] += 1;
                return digits;
            }
            // if 9, make 0, then increment next
            digits[index] = 0;   
        }

        // if all nines, no matter how many
        int[] newArray = new int[digits.length + 1];
        newArray[0] = 1;
      
        return newArray;

    }
}